### PR TITLE
Plutus modules restructure

### PR DESCRIPTION
**Pull request by lehins**
# Description

This PR is the first stage in untangling Plutus mess. Initial step for #3849 

There are no semantic changes in this PR. It only organizes the location of the code. All of the Plutus specific functionality that is era agnostic has been moved into core and organized logically into separate modules. All of git history is preserved, despite what the github diff portrays, which is important that the commits are not squashed together, because history will be lost if that happens.

Here is the re-organization:

* Move `Cardano.Ledger.Language` to `Cardano.Ledger.Plutus.Language`
* Move `Cardano.Ledger.Alonzo.Scripts.Data` to `Cardano.Ledger.Plutus.Data`
* Extract `ExUnits` and `Prices` functionality from `Cardano.Ledger.Alonzo.Scripts` into `Cardano.Ledger.Plutus.ExUnits` module.
* Extract `CostModel` and `CostModels` functionality from `Cardano.Ledger.Alonzo.Scripts` into `Cardano.Ledger.Plutus.CostModels` module.
* Move script evaluation functionality from `Cardano.Ledger.Alonzo.TxInfo` into `Cardano.Ledger.Plutus.Evaluate`
* Move era independent plutux context translation functionality from `Cardano.Ledger.Alonzo.TxInfo` into `Cardano.Ledger.Plutus.TxInfo`
 
# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

**Review by Soupstraw — APPROVED** [review]

File: libs/cardano-ledger-core/cardano-ledger-core.cabal [hunk 1]
```diff
@@ -47,11 +47,12 @@ library
         Cardano.Ledger.Keys
         Cardano.Ledger.Keys.Bootstrap
         Cardano.Ledger.Keys.WitVKey
-        Cardano.Ledger.Language
+        --Cardano.Ledger.Language
```
Comment (teodanciu) [inline libs/cardano-ledger-core/cardano-ledger-core.cabal]: ~Left by accident I think~
Nevermind, I see what happened in the next commit.

File: eras/conway/impl/cardano-ledger-conway.cabal [hunk 2]
```diff
@@ -78,9 +78,9 @@ library
         cardano-data >=1.1.2.0,
         cardano-ledger-binary >=1.2,
         cardano-ledger-allegra >=1.1,
-        cardano-ledger-alonzo ^>=1.5,
+        cardano-ledger-alonzo ^>=1.5.1,
```
Comment (teodanciu) [inline eras/conway/impl/cardano-ledger-conway.cabal:81]: I think this requires a version bump of this package to 1.10.1.1

File: libs/cardano-ledger-api/cardano-ledger-api.cabal [hunk 3]
```diff
@@ -55,11 +55,11 @@ library
         aeson >=2.2,
         array,
         cardano-ledger-allegra >=1.1,
-        cardano-ledger-alonzo >=1.4,
+        cardano-ledger-alonzo >=1.5.1,
```
Comment (teodanciu) [inline libs/cardano-ledger-api/cardano-ledger-api.cabal:58]: And this too should cause incrementing the version of the package to 1.7.0.1, no?

**Review by teodanciu — APPROVED** [review]
Just a few questions about version adjustments. 
Awesome git discipline 🤩
Looks good to me.

File: libs/cardano-ledger-api/cardano-ledger-api.cabal (hunk 3 above)
Comment (lehins) [reply libs/cardano-ledger-api/cardano-ledger-api.cabal:58]: Thank you for catching this. I though I got them all.

File: eras/conway/impl/cardano-ledger-conway.cabal (hunk 2 above)
Comment (lehins) [reply eras/conway/impl/cardano-ledger-conway.cabal:81]: This actually has to be `1.11.0.0`, it was only bumped in the changelog before, but not in the cabal file. Thank you for pointing me at this as well!

**Pull request merged**